compiler/qsc_parse/src/expr/tests.rs

```diff
@@ -2181,3 +2181,197 @@ fn nested_interpolated_string_with_exprs() {
                 Lit: " baz""#]],
     );
 }
+
+#[test]
+fn duplicate_commas_in_tuple() {
+    check(
+        expr,
+        "(x,, y)",
+        &expect![[r#"
+            Expr _id_ [0-7]: Tuple:
+                Expr _id_ [1-2]: Path: Path _id_ [1-2] (Ident _id_ [1-2] "x")
+                Expr _id_ [0-0]: Err
```

Comment:
Haven't looked at the rest of the code yet but I really think these error nodes should have accurate spans. They can be 0-length spans if necessary. That would be consistent with the behavior of other recovering parsers. You also need accurate spans in language service features to determine which AST node the cursor falls into.

Reply:
Ah, that was exactly the source of the issue I was seeing with signature help! Thanks for pointing that out. I was able to add a trait that updates the spans such that the right zero-length span shows up in the Err AST members and now signature help behaves as expected.